cfdm/test/test_read_write.py

```diff
@@ -172,10 +177,174 @@ def test_read_write_format(self):
                 f.equals(g, verbose=3), f"Bad read/write of format: {fmt}"
             )
 
+    def test_write_netcdf_mode(self):
+        """Test the `mode` parameter to `write`, notably append mode."""
+        g = cfdm.read(self.filename)  # note g has one field
+        g_copy = g.copy()
+
+        for fmt in self.netcdf_fmts:  # test over all netCDF 3 and 4 formats
+            # Other tests cover write as default mode (i.e. test with no mode
+            # argument); here test explicit provision of 'w' as argument:
+            cfdm.write(g, tmpfile, fmt=fmt, mode="w")
+            f = cfdm.read(tmpfile)
+
+            new_length = 1  # since 1 == len(g)
+            self.assertEqual(len(f), new_length)
+            self.assertTrue(f[0].equals(g[0]))
+
+            # Main aspect of this test: testing the append mode ('a'): now
+            # append all other example fields, to check a diverse variety.
+            for ex_field_n, ex_field in enumerate(cfdm.example_fields()):
+                # Note: after Issue #141, this skip can be removed.
+                if ex_field_n == 1:
+                    continue
+                # Skip since "Can't write int64 data from <Count: (2) > to a
+                # NETCDF3_CLASSIC file" causes a ValueError i.e. not possible.
+                # Note: can remove this when Issue #140 is closed.
+                if fmt in self.netcdf3_fmts and ex_field_n == 6:
+                    continue
+
+                cfdm.write(ex_field, tmpfile, fmt=fmt, mode="a")
```

Comment:
I'm getting a failure here for `fmt=NETCDF4_CLASSIC` and `n=3`:

```
  File "/home/david/cfdm/cfdm/read_write/netcdf/netcdfwrite.py", line 270, in _write_attributes
    g["nc"][ncvar].setncatts(netcdf_attrs)

AttributeError: NetCDF: Attempt to define fill value when data already exists.
```

This looks to me like an unrelated problem getting trapped at this low level trap. Do you see this?

The field is a DSG - could that be part of it? although I can't see how that would affect things in this way ....

Reply:
Ah, important find there, good job; I have seen that one before, but thought recent changes had fixed it as I hadn't seen it for a while.

The error message would indeed suggest an unrelated problem, though regardless of whether or not it is related to this new code we should get it fixed for the imminent release, I think. I'm looking into it...

Reply:
Should now be fixed...